File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/library/items/playlists/playback/IPlaylistPlayer.java
package com.lasthopesoftware.bluewater.client.library.items.playlists.playback;

import com.lasthopesoftware.bluewater.client.library.items.media.files.playback.file.PositionedPlaybackFile;
import com.lasthopesoftware.bluewater.shared.IUsefulObservable;

import java.io.Closeable;

/**
 * Created by david on 11/7/16.
 */

public interface IPlaylistPlayer extends IUsefulObservable<PositionedPlaybackFile>, Closeable {
	void pause();
	void resume();

	void setVolume(float volume);

	void cancel();
}

File: playback/src/main/java/com/namehillsoftware/projectblue/playback/playlist/IPlaylistPlayer.kt
package com.lasthopesoftware.bluewater.client.playback.playlist

import com.namehillsoftware.handoff.promises.Promise
import com.namehillsoftware.projectblue.playback.file.PositionedPlayingFile
import io.reactivex.ObservableOnSubscribe

/**
 * Created by david on 11/7/16.
 */
interface IPlaylistPlayer : ObservableOnSubscribe<PositionedPlayingFile> {
	fun pause(): Promise<*>
	fun resume(): Promise<PositionedPlayingFile?>
	fun setVolume(volume: Float)
	val isPlaying: Boolean
}
